### PR TITLE
storage, config: Clean up backends.go

### LINKED ISSUE #382: backend.NewDriver needs some TLC
This call takes a lot of differing arguments that change the behavior of it dramatically.

Suggestion: use a struct as the argument, which essentially gives us named parameters (that we can choose to ignore).

Either way this is going to require a fair bit of cleanup in the `api/` module alongside it.

### PR DESCRIPTION
**Pull request by yuva29**
Fixes #382

Signed-off-by: Yuva Shankar <yuva.shankar29@gmail.com>

### CODE REVIEW

**Comment by erikh** [comment]
is this ready for review? CI failed.

**Comment by yuva29** [comment]
Yes, it is. CI failed due to Etcd.

File: config/volume.go [hunk 1]
```diff
@@ -353,31 +353,47 @@ func (cfg *Volume) Validate() error {
...
-		if err := backend.NewDriver(backendName, driverType, mountPath, &do); err != nil {
+	mnt, err := backend.NewMountDriver(cfg.Backends.Mount, backend.MountPath)
+	if err != nil {
+		return err
+	}
+
+	if err := mnt.Validate(&do); err != nil {
+		return err
+	}
+	if cfg.Backends.Snapshot != "" {
```
Comment (dseevr) [inline config/volume.go]: Pedantic:  should have a blank line between these two if statement blocks 😄

File: config/volume.go [hunk 2]
```diff
@@ -353,31 +353,47 @@ func (cfg *Volume) Validate() error {
...
+		snapshot, err := backend.NewSnapshotDriver(cfg.Backends.Snapshot)
+		if err != nil {
 			return err
 		}
-	}
+		if err := snapshot.Validate(&do); err != nil {
```
Comment (dseevr) [inline config/volume.go]: More pedantry:  lines 392 and 393 should be swapped

**Comment by dseevr** [comment]
LGTM

**Comment by erikh** [comment]
LGTM

**Pull request merged**